orchestrator/test_runner/src/happy_path_v2.rs

```diff
@@ -64,33 +60,40 @@ pub async fn happy_path_test_v2(
         "Successfully deployed new ERC20 representing FooToken on Cosmos with event nonce {}",
         ending_event_nonce
     );
-
-    let start = Instant::now();
     // the erc20 representing the cosmos asset on Ethereum
     let mut erc20_contract = None;
-    while Instant::now() - start < TOTAL_TIMEOUT {
-        let res = grpc_client
-            .denom_to_erc20(DenomToErc20Request {
-                denom: token_to_send_to_eth.clone(),
-            })
-            .await;
-        if let Ok(res) = res {
+    match tokio::time::timeout(TOTAL_TIMEOUT, async {
+        loop {
+            let res = grpc_client
+                .denom_to_erc20(DenomToErc20Request {
+                    denom: token_to_send_to_eth.clone(),
+                })
+                .await;
+
+            if let Ok(res) = res {
+                break res;
+            }
+            tokio::time::sleep(std::time::Duration::from_secs(1)).await;
+        }
+    })
+    .await
+    {
+        Ok(res) => {
             let erc20 = res.into_inner().erc20;
+            erc20_contract = Some(erc20);
             info!(
-                "Successfully adopted {} token contract of {}",
-                token_to_send_to_eth, erc20
+                "Successfully adopted {} token contract",
+                token_to_send_to_eth
```

Comment:
We're capturing `erc20` now but still no longer printing it in the success message, as in the original code.

Reply:
I missed it, this commit fixes it:

https://github.com/PeggyJV/gravity-bridge/pull/274/commits/9b958a32c161ed93571944d96d9bf1cde976a227